src/leveled_codec.erl

```diff
@@ -468,16 +468,16 @@ isvalid_ledgerkey(_LK) ->
 %% false and further results may be required from further ranges.
 endkey_passed(all, _) ->
     false;
-endkey_passed({K1, null, null, null}, {K1, _, _, _}) ->
-    false;
-endkey_passed({K1, K2, null, null}, {K1, K2, _, _}) ->
-    false;
-endkey_passed({K1, K2, K3, null}, {K1, K2, K3, _}) ->
-    false;
-endkey_passed({K1, null}, {K1, _}) ->
+endkey_passed({KQ1, null, null, null}, {KR1, _, _, _}) when KQ1 =/= null ->
```

Comment:
be aware that:
```
null < <<"a">>.
true
```
Thus `endkey_passed({nul, null, null, null}, {null, null, null, <<"a">>})`  drops through to the last clause and results in `true` whereas in the old version it would return a `false`.

Reply:
null means don't care in this case, so if we don't care about all elements of the tuple, then we should use `all`, which will return a false because of the first clause.